spec/controllers/accounts_controller_spec.rb

```diff
@@ -8,6 +8,65 @@
   let(:account) { Fabricate(:account) }
 
   describe 'GET #show' do
+    context 'with basic account status checks' do
+      shared_examples 'unapproved account check' do
+        before { account.user.update(approved: false) }
+
+        it 'returns http not found' do
+          get :show, params: { username: account.username, format: format }
+
+          expect(response).to have_http_status(404)
+        end
+      end
+
+      shared_examples 'permanently suspended account check' do
+        before do
+          account.suspend!
+          account.deletion_request.destroy
+        end
+
+        it 'returns http gone' do
+          get :show, params: { username: account.username, format: format }
+
+          expect(response).to have_http_status(410)
+        end
+      end
+
+      shared_examples 'temporarily suspended account check' do |code: 403|
+        before { account.suspend! }
+
+        it 'returns http forbidden' do
+          get :show, params: { username: account.username, format: format }
+
+          expect(response).to have_http_status(code)
+        end
```

Comment:
That description sounds misleading as `code` is variable and sometimes 200.

Reply:
Updated wording to be less specific.